### DISCUSSION TITLE
Does IDOM support all React components?

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#358] [Question] [answered]
The docs mention react components can be used. Are there limitations?

**Comment by rmorshea** [comment, marked as answer]
With respect to components installed with `$ idom install` or `idom.install()` there's definitely some limitations that, unfortunately, I haven't really had the time to document yet. In short, all component props have to be JSON serializable, so that's a problem if you need to assign a JS object to a prop. This is a common pattern in Material UI so it's come up a number of times in that context.

That probably won't change though because there is an [interface](https://idom-docs.herokuapp.com/docs/javascript-components.html#import-javascript-bundles) for integrating your own custom components in JS and distributing them as a Python package. However, I have plans to [reconsider that soon](https://idom-docs.herokuapp.com/docs/roadmap.html) so I've been hesitant to suggest people use it.
